maven-model-transform/src/test/java/org/apache/maven/model/transform/ConsumerPomXMLFilterTest.java

```diff
@@ -226,4 +235,40 @@ void lexicalHandler() throws Exception {
         String actual = transform(input);
         assertThat(actual).and(expected).areIdentical();
     }
+
+    @Test
+    void testWithEntities() throws Exception {
+        String input = "<!DOCTYPE foo [\n"
+                + "  <!ENTITY desc SYSTEM \"file:desc.xml\">\n"
+                + "]>\n"
+                + "<project>"
+                + "&desc;"
+                + "</project>";
+        String expected = "<project>" + "<description>foo</description>" + "</project>";
+
+        XMLResolver resolver = new XMLResolver() {
+            @Override
+            public Object resolveEntity(String publicID, String systemID, String baseURI, String namespace)
+                    throws XMLStreamException {
+                if ("file:desc.xml".equals(systemID)) {
+                    return "<?xml version='1.0' encoding='UTF-8'?><description>foo</description>";
+                }
+                return null;
+            }
+        };
+        WstxInputFactory factory = new WstxInputFactory();
+        factory.setProperty(XMLInputFactory.IS_REPLACING_ENTITY_REFERENCES, true);
+        factory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, true);
+        factory.setProperty(WstxInputProperties.P_TREAT_CHAR_REFS_AS_ENTS, false);
+        factory.setProperty(WstxInputProperties.P_UNDECLARED_ENTITY_RESOLVER, resolver);
+        factory.setProperty(WstxInputProperties.P_ENTITY_RESOLVER, resolver);
+        XMLStreamReader parser = factory.createXMLStreamReader(new StringReader(input));
+        XMLStreamReader filter = getFilter(parser);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        XmlUtils.writeDocument(filter, baos);
+        String actual = baos.toString();
+
+        System.out.println(actual);
```

Comment:
No println in tests please. Only failing tests should generate output so failures are easier to spot and debug